openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -276,6 +280,13 @@ def run(self, problem):
             # Use pyOpt's internal finite difference
             fd_step = problem.root.fd_options['step_size']
             sol = opt(opt_prob, sens='FD', sensStep=fd_step, storeHistory=self.hist_file)
+        elif self.options['snopt_diff']:
+            if self.options['optimizer']=='SNOPT':
+                fd_step = problem.root.fd_options['step_size']
+                sol = opt(opt_prob, sens=None, sensStep=fd_step, storeHistory=self.hist_file)
+            else:
+                msg = "SNOPT's internal finite difference method can only be used with SNOPT"
+                raise Exception(msg)
```

Comment:
The only thing I'd add is one more test where we set snopt_diff to True, but use one of the other optimizers so that we get test coverage on the error message. Aside from that small thing, this pull request looks pretty solid.